modules/Bio/EnsEMBL/IO/Parser/BaseVCF4.pm

```diff
@@ -655,31 +656,54 @@ sub get_format_description {
 }
 
 
-=head2 get_metadata_description
+=head2 get_metadata_field
     Argument [1]: Metadata type, e.g. 'INFO'
     Argument [2]: Metadata ID, e.g. 'AA'
-    Description : Retrieve the description of the given metadata type and metadata ID
+    Argument [3]: Metadata field, e.g. 'Description'
+    Description : Retrieve a field of the given metadata type and metadata ID
     Returntype  : String
 =cut
 
-sub get_metadata_description {
-  my $self = shift;
-  my $type = shift;
-  my $id   = shift;
+sub get_metadata_field {
+  my $self  = shift;
+  my $type  = shift;
+  my $id    = shift;
+  my $field = shift;
 
   if (!defined($type) || !defined($id)) {
     carp("You need to provide a meta type (e.g. 'INFO') and a meta entry ID (e.g. 'AA')");
     return;
+  } elsif (!defined($field)) {
+    carp("You need to provide a field (e.g. 'Description')");
+    return;
   }
 
   my $meta = $self->get_metadata_by_pragma($type);
   foreach my $meta_entry (@$meta) {
-    return $meta_entry->{'Description'} if ($meta_entry->{'ID'} eq $id);
+    if ($meta_entry->{'ID'} eq $id and $meta_entry->{$field}) {
+      return $meta_entry->{$field};
+    }
   }
   return;
 }
 
 
+=head2 get_metadata_description
+    Argument [1]: Metadata type, e.g. 'INFO'
+    Argument [2]: Metadata ID, e.g. 'AA'
+    Description : Retrieve the description of the given metadata type and metadata ID
+    Returntype  : String
+=cut
+
+sub get_metadata_description {
+  my $self = shift;
+  my $type = shift;
+  my $id   = shift;
+
+  return $self->get_metadata_field($type, $id, 'Description');
```

Comment:
I don't like this call to the new sub as the return field for the existing sub. 1) I think this sub should be independent, 2) I think it will fail the check for field in get_metadata_field, thus breaking existing functionality.

Suggest changing this line to the code originally in this sub:
```suggestion
  if (!defined($type) || !defined($id)) {
    carp("You need to provide a meta type (e.g. 'INFO') and a meta entry ID (e.g. 'AA')");
    return;
}

  my $meta = $self->get_metadata_by_pragma($type);
  foreach my $meta_entry (@$meta) {
    return $meta_entry->{'Description'} if ($meta_entry->{'ID'} eq $id);
}
return;
```
I'm tagging @sarahhunt as I think you discussed this change with Nuno?

Reply:
Hi @jamie-m-a, thanks! Although the function was passing the unit tests as expected, I agree with leaving the original code to avoid potential untested changes. Reverted!